src/charm.py

```diff
@@ -872,16 +874,40 @@ def enable_disable_extensions(self, database: str = None) -> None:
                 for ext in spi_module:
                     extensions[ext] = enable
                 continue
-            if extension in plugins_exception:
-                extension = plugins_exception[extension]
+            extension = plugins_exception.get(extension, extension)
+            if self._check_extension_dependencies(extension, enable):
+                self.unit.status = BlockedStatus(EXTENSIONS_BLOCKING_MESSAGE)
```

Comment:
We should define a different message constant. Something like `unsatisfied plugin dependencies. Please check the logs`